File: results-archive/fastcrc/DeepSeekV4Flash-Vision-EXP-ablit-pi-20260902-1/src/crc.ts
/**
 * CRC-32 (IEEE 802.3 / ISO-HDLC): reflected polynomial 0xEDB88320,
 * init 0xFFFFFFFF, final XOR 0xFFFFFFFF. Pure — BCL/global objects only.
 */
const TABLE: Uint32Array = (() => {
  const t = new Uint32Array(256);
  for (let n = 0; n < 256; n++) {
    let c = n;
    for (let k = 0; k < 8; k++) {
      c = (c & 1) !== 0 ? 0xedb88320 ^ (c >>> 1) : c >>> 1;
    }
    t[n] = c >>> 0;
  }
  return t;
})();

export function crc32(data: Uint8Array): number {
  let c = 0xffffffff;
  for (let i = 0; i < data.length; i++) {
    c = (TABLE[(c ^ data[i]!) & 0xff]! ^ (c >>> 8)) >>> 0;
  }
  return (c ^ 0xffffffff) >>> 0;
}

File: results/fastcrc/GLM-5.3-Flash-OpenRouter-pi-20260902-2/src/Fastcrc/Crc.cs
namespace Fastcrc;

/// <summary>Pure CRC-32 (IEEE 802.3) computation — the only algorithm module.</summary>
public static class Crc
{
    /// <summary>Reflected polynomial 0xEDB88320 of 0x04C11DB7.</summary>
    private const uint Poly = 0xEDB88320u;

    private static readonly uint[] Table = BuildTable();

    private static uint[] BuildTable()
    {
        var table = new uint[256];
        for (uint i = 0; i < 256; i++)
        {
            var crc = i;
            for (var bit = 0; bit < 8; bit++)
                crc = (crc & 1) != 0 ? (crc >> 1) ^ Poly : crc >> 1;
            table[i] = crc;
        }
        return table;
    }

    /// <summary>Computes the CRC-32 checksum of <paramref name="data"/> (init 0xFFFFFFFF, xorout 0xFFFFFFFF).</summary>
    public static uint Crc32(byte[] data)
    {
        var crc = 0xFFFFFFFFu;
        foreach (var b in data)
            crc = (crc >> 8) ^ Table[(crc ^ b) & 0xFF];
        return crc ^ 0xFFFFFFFFu;
    }
}
